code/datums/keybinding/robot.dm

```diff
@@ -64,3 +64,22 @@
 	var/mob/living/silicon/robot/R = user.mob
 	R.uneq_active()
 	return TRUE
+
+/datum/keybinding/robot/undeploy
+	category = CATEGORY_AI
+	hotkey_keys = list("=")
+	name = "undeploy"
+	full_name = "Disconnect from shell"
+	description = "Returns you to your AI core"
+	keybind_signal = COMSIG_KB_SILION_UNDEPLOY_DOWN
+
+/datum/keybinding/robot/undeploy/down(client/user)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/silicon/robot/shell/our_shell = user.mob
```

Comment:
This can be used while not actually inside of a shell

Reply:
That's what the check in the next couple lines are for, if that's not good enough as stated above I could implement an IsAIShell() check and test that right at the start of down()? When I tested this as a cyborg on local it didn't spit any runtimes at me.

Reply:
Reading over this again, this is fine and should work.